### DISCUSSION TITLE
Running pocketbase on a powerful router with openwrt

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2552] [Show and tell]
Does anyone know if Pocketbase could target an openwrt router?  I'm shopping around, and wondering if it could just be easier and better to make a roaming 802.11r or mesh 802.11s network and to just use the router itself as the server instead of having extra Raspberry Pi like devices alongside it for home / farm automation.  It seems like that Pocketbase would be pretty ideal for this.  Maybe.

I found this openwrt router that is available locally.  It has a decent processor and 1GB of memory.  There are others with MediaTek processors if it's not a good fit.

Router Wi-Fi Xiaomi Mi Router AX9000
Qualcomm Atheros IPQ8072A 1024Mhz
256 flash
1024 mb memory

**Comment by dschissler** [comment]
@ganigeorgiev Also I know that you're not in extreme optimization mode yet, but getting the web assets down with brotli could be a way to squeeze onto more devices.

**Comment by dschissler** [comment]
I figured out that *probably* a Linksys WRT3200ACM with OpenWRT router would be powerful enough to run Pocketbase for something light like home automation.  There are many OpenWRT routers with weak MIPS CPUs that wouldn't be compatible nor fast enough.  I see that OpenWRT has an nginx package.  So just put it behind a reverse proxy to the PB program running directly on the router.  With this setup the PB server program could be easily migrated to its own box later.  Just zip up the data directory, recompile, and change the nginx vhost config to send traffic to the local LAN IP (new box).  It's very flexible, and possibly saves the need to have another $50-200 box hanging around.

I spent a fair amount of time setting up a [Wish](https://github.com/charmbracelet/wish) SSH server and client.  So you could use this as a front facing portal that sends signals to other devices by SSH.  Again I like Pocketbase here because even a PB server as Framework without a web server still gets the Pocketbase Database environment.  So why not just use Pocketbase for everything?  If I ever need even a single table then it's already right there.

**Comment by dschissler** [comment]
I'll add that this powerful router came out over five years ago.  The next iterations of comparably powerful hacker routers should be more than enough to run Pocketbase with private use without even thinking about it.

**Comment by dekarl** [comment]
FWIW I just tried building [opensoho](https://github.com/rubenbe/opensoho) to run on Realtek based switches, which are mipsel.softtfloat like the Xiaomi Mi routers. [The missing part to "just build it and try it" is mipsel support in modernc.org/libc modernc.org/sqlite.](https://pkg.go.dev/modernc.org/sqlite#hdr-Supported_platforms_and_architectures)

OTOH for Home/Farm Automation, just use a small dedicated device running HomeAssistant and build your custom stuff around that.
